FastPriorityQueue.js

```diff
@@ -287,13 +287,22 @@ FastPriorityQueue.prototype.forEach = function(callback) {
 FastPriorityQueue.prototype.kSmallest = function(k) {
   if (this.size == 0) return [];
   k = Math.min(this.size, k);
-  var fpq = new FastPriorityQueue(this.compare);
-  const newSize = Math.min((k > 0 ? Math.pow(2, k - 1) : 0) + 1, this.size);
+  const newSize = Math.min(this.size, (1 << (k - 1)) + 1);
+  if (newSize < 2) {
+    if (newSize < 1)  {
```

Comment:
If you add the guard above, then newSize cannot be zero because this.size>0, and (1<<(k-1)+1) is at least 2. So this branch would not be needed.

You could just say `if(newSize == 1) { return [this.peek()]; }`.

Reply:
I found `newSize == 1` appeared to reduce performance, so I went with `newSize < 2`. This will be safe because `this.size` and `(1 << (k - 1)) + 1` are both guaranteed to be `>= 1` when this is checked (excepting cases that were also previously unsupported).